### PR TITLE
MSEARCH-86: Increase retry period.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
Increase retry period to 20 seconds

### CODE REVIEW

**Review by hjiebsco — APPROVED** [review]
Just be sure, can you verify this handles gracefully if someone manually deletes/expires messages using Kafka tools?

**Review by pfilippovEPAM — APPROVED** [review]

File: src/main/java/org/folio/search/configuration/KafkaConfiguration.java [hunk 1]
```diff
@@ -43,9 +47,17 @@
...
     factory.setBatchListener(true);
     factory.setConsumerFactory(jsonNodeConsumerFactory());
+    factory.setBatchErrorHandler(kafkaFactoryErrorHandler());
     return factory;
   }
 
+  private BatchErrorHandler kafkaFactoryErrorHandler() {
+    var errorHandler = new SeekToCurrentBatchErrorHandler();
+    // infinity retry every 20 seconds
+    errorHandler.setBackOff(new FixedBackOff(20_000, UNLIMITED_ATTEMPTS));
```
Comment (pfilippovEPAM) [inline src/main/java/org/folio/search/configuration/KafkaConfiguration.java:57]: everything looks good, except this line, maybe we will extract this property to application.yml?

Comment (bohdan-suprun) [reply src/main/java/org/folio/search/configuration/KafkaConfiguration.java]: Make sense, extracted.

**Comment by bohdan-suprun** [comment]
> Just be sure, can you verify this handles gracefully if someone manually deletes/expires messages using Kafka tools?

You're right, It seems I misunderstood documentation. If we set UNLIMMITTED renewals than producer will keep records in memory and redeliver without syncing metadata with Kafka broker.

I have set the attempts count to 10 so I believe it should retry 10 times and then update offset for topic so that the records will be delivered again but within another attempt (when pool() is called next time).

I'll double check it on Rancher tomorrow.

**Comment by bohdan-suprun** [comment]
Verified, works as expected on Rancher

**Pull request merged**